example/open-new-position-lp.ts

```diff
@@ -0,0 +1,111 @@
+// Always set the ENV as 'prod' before importing the package.
+// You can also consider using `dotenv` package to make this a bit cleaner.
+process.env.ENV = 'prod'
+
+import {
+  Vault__factory,
+  IWorker__factory,
+  BEP20__factory,
+} from '@alpaca-finance/alpaca-contract/typechain'
+import { AddAllBaseTokenStrategy, TwoSideOptimalStrategy } from '@alpaca-finance/alpaca-sdk/build-cjs/entity/Strategy'
+import { formatEther } from '@ethersproject/units'
+import { ethers } from 'hardhat'
+import { BigNumber, constants } from 'ethers'
+
+async function main() {
+  const OpenPositionGasLimit = BigNumber.from('2700000')
+  const signer = (await ethers.getSigners())[0]
+  console.log('signer', signer.address)
+
+ // _____                               _                
+ // |  __ \                             | |               
+ // | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
+ // |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
+ // | |  | (_| | | | (_| | | | | | |  __/ ||  __/ |  \__ \
+ // |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/
+                                                        
+  const vaultAddress = '0xe5ed8148fE4915cE857FC648b9BdEF8Bb9491Fa5' // BUSD Vault
+  const workerAddress = '0x8164104FeaA27Ac52d7CD22D63a5Ef50971cB7ff' // ALPACA-BUSD PancakeswapWorker
+  const strategyAddress = '0x22fC6110d5d9b122f3C2d4715C24566342161a12' // ALPACA-BUSD PancakeswapWorker.strategies.StrategyAddTwoSidesOptimal
```

Comment:
Is this address for [PancakeswapV2StrategyAddTwoSidesOptimal](https://github.com/alpaca-finance/bsc-alpaca-contract/blob/main/contracts/6/protocol/strategies/pancakeswapV2/PancakeswapV2StrategyAddTwoSidesOptimal.sol) or for [PancakeswapV2RestrictedStrategyAddTwosidesOptimal](https://github.com/alpaca-finance/bsc-alpaca-contract/blob/main/contracts/6/protocol/strategies/pancakeswapV2-restricted/PancakeswapV2RestrictedStrategyAddTwosidesOptimal.sol)?

Reply:
I just opened a real position using [Alpaca Finance frontend on testnet](https://testnet-lfg-app.alpacafinance.org/farm) and here is the [transaction](https://testnet.bscscan.com/tx/0xc7beeab4ee89fadca4d0684bb3fbcf1c6dc44cc2a0507a21688fc787a70b3b7c).

From the transaction input data it seems the strategy is still the old one: PancakeswapV2StrategyAddTwoSidesOptimal. See input [8] and [9], those should be two addresses, right?
![image](https://user-images.githubusercontent.com/1639620/146156551-757074ac-e895-421d-90cb-2125e3045fc7.png)

Is your test case working now?

Reply:
If this transaction is executing on the new strategy, baseTokenAddress and farmingTokenAddress should not be here... Then what does [8] and [9] mean?

Reply:
> Is this address for [PancakeswapV2StrategyAddTwoSidesOptimal](https://github.com/alpaca-finance/bsc-alpaca-contract/blob/main/contracts/6/protocol/strategies/pancakeswapV2/PancakeswapV2StrategyAddTwoSidesOptimal.sol) or for [PancakeswapV2RestrictedStrategyAddTwosidesOptimal](https://github.com/alpaca-finance/bsc-alpaca-contract/blob/main/contracts/6/protocol/strategies/pancakeswapV2-restricted/PancakeswapV2RestrictedStrategyAddTwosidesOptimal.sol)?

This is the address for PancakeswapV2RestrictedStrategyAddTwosidesOptimal. I have verified the contract in testnet which you can view the implementation code here https://testnet.bscscan.com/address/0xda6031c074d41a28df8827b3cf6b66352ea3ba7a#readProxyContract

Reply:
> If this transaction is executing on the new strategy, baseTokenAddress and farmingTokenAddress should not be here... Then what does [8] and [9] mean?

[8] and [9] seem to be just a placeholder of some sort. I am not sure about this but they do not matter in our context. The ones you need to look for are [10] and [11] instead.

[10] is `farmingTokenAmount` where the value `3635c9adc5dea00000` in base16 equals to `1000000000000000000000` or 100 BUSD which is consistent with the 100 BUSD transfer in the transaction.

[11] is `minLpAmount` which is a slippage control calculated from the frontend. The value `1c4e918b51a1a28389` in base16 equals to `522170293453169263497` or 522.1702934531693 minimum LP token. This transaction passed the `minLpAmount` check by producing 523.479651593504030007 LP token which is larger than the `minLpAmount`.

Reply:
Yes thanks to explain! Our understanding on [10] and [11] are the same.